src/string_id.h

```diff
@@ -171,24 +192,14 @@ class string_id
             return !is_null();
         }
 
-        // TODO: Exposed for now. Hide these and make them accessible to the generic_factory only
-
-        /**
-         * Assigns a new value for the cached int id.
-         */
-        void set_cid( const int_id<T> &cid ) const {
-            _cid = cid.to_i();
-        }
```

Comment:
Maybe we should keep `set_cid` as a private function that takes two arguments (id and version)?  The two values are always set together.